polygeist/tools/cgeist/Test/Verification/sycl/structvec.cpp

```diff
@@ -0,0 +1,77 @@
+// RUN: clang++ -fsycl -fsycl-device-only -O0 -w -emit-mlir %s -o - | FileCheck %s
+
+#include <sycl/sycl.hpp>
+#include <initializer_list>
+
+struct structvec {
+  using char2 = char __attribute__((ext_vector_type(2)));
+  char2 v;
+
+  structvec(std::initializer_list<char> l) {
+    for (unsigned I = 0; I < 2; ++I) {
+      v[I] = *(l.begin() + I) ? -1 : 0;
+    }
+  }
+};
+
+// CHECK-LABEL: func.func @_Z10test_store9structvecic(%arg0: !llvm.ptr<struct<(vector<2xi8>)>> {llvm.align = 2 : i64, llvm.byval = !llvm.struct<(vector<2xi8>)>, llvm.noundef}, %arg1: i32 {llvm.noundef}, %arg2: i8 {llvm.noundef, llvm.signext}) -> !llvm.struct<(vector<2xi8>)>
+// CHECK-NEXT:    %c1_i64 = arith.constant 1 : i64
+// CHECK-NEXT:    %0 = llvm.alloca %c1_i64 x !llvm.struct<(vector<2xi8>)> : (i64) -> !llvm.ptr<struct<(vector<2xi8>)>>
+// CHECK-NEXT:    %1 = llvm.addrspacecast %0 : !llvm.ptr<struct<(vector<2xi8>)>> to !llvm.ptr<struct<(vector<2xi8>)>, 4>
```

Comment:
`mem2reg` should be handling this. It's a similar issue to #7645. I'll handle it when this is merged :+1: